docs/CONFIG.md

```diff
@@ -0,0 +1,117 @@
+## Introduction
+Envoy Gateway consists of two types of configuration, static and dynamic. Static is used to configure Envoy Gateway at
+runtime and dynamic configuration is used to manage the Envoy proxies. This document provides Envoy Gateway
+configuration details.
+
+## Static Config
+This is the configuration used to configure various internal aspects of Envoy Gateway at runtime.
+
+#### Configuration File
+The configuration file is defined by the ControlPlaneSpec API type. At startup, Envoy Gateway searches for the
+configuration at "/etc/envoy-gateway/config.yaml". You can override the path to this file with `-c <path string>` or
+`--config-path <path string>`. For example:
+```shell
+envoy-gateway start -c config.yaml
+```
+
+#### CLI Arguments
+If no value is provided for a given option, a default value applies. If an option has sub-options, and any of these
+sub-options are not specified, a default value will also apply. For example, the `--providers.kubernetes` option will
```

Comment:
prefer if we dont start supporting all these options, becomes hard to maintain them, instead prefer if these are provided
in the config spec

Reply:
@arkodg thanks for the feedback. I was going back and forth on whether to include CLI and env vars. For now, I'm going to proceed with EG only supporting config through a file. We can add CLI args and env vars in the future when needed.